src/java/arjdbc/postgresql/PGDriverImplementation.java

```diff
@@ -190,8 +214,17 @@ public boolean setStringParameter(final ThreadContext context,
             if ( sqlType != null ) {
                 if ( PostgreSQLRubyJdbcConnection.rawArrayType == Boolean.TRUE && arrayLike(sqlType) ) {
                     final int oid = PostgreSQLRubyJdbcConnection.oid(context, column);
-                    Jdbc4Array valueArr = new Jdbc4Array(connection.unwrap(BaseConnection.class), oid, valueStr);
-                    statement.setArray(index, valueArr); return true;
+                    final Class<?> clazz = pgArrayClass(context);
+                    try {
+                        Array valueArr = (Array) clazz
```

Comment:
this is good ... but the driver we're compiling against should be implemented directly with a reflection fallback if using and older driver. detection should happen once (e.g. in a static final flag).